File: legacy/src/main/java/com/xiaomi/mipush/sdk/AssemblePushInfoHelper.java
package com.xiaomi.mipush.sdk;

import com.xiaomi.xmpush.thrift.ConfigKey;
import java.util.HashMap;

/* JADX INFO: loaded from: miuipushsdkshared_3_7_9.jar:com/xiaomi/mipush/sdk/AssemblePushInfoHelper.class */
public class AssemblePushInfoHelper {
    protected static final String COS_PUSH_MANAGER_CLASS_NAME = "com.xiaomi.assemble.control.COSPushManager";
    protected static final String COS_PUSH_SINGLE_METHOD_NAME = "newInstance";
    protected static final String FCM_PUSH_MANAGER_CLASS_NAME = "com.xiaomi.assemble.control.FCMPushManager";
    protected static final String FCM_PUSH_SINGLE_METHOD_NAME = "newInstance";
    protected static final String FTOS_PUSH_MANAGER_CLASS_NAME = "com.xiaomi.assemble.control.FTOSPushManager";
    protected static final String FTOS_PUSH_SINGLE_METHOD_NAME = "newInstance";
    protected static final String HMS_PUSH_MANAGER_CLASS_NAME = "com.xiaomi.assemble.control.HmsPushManager";
    protected static final String HMS_PUSH_SINGLE_METHOD_NAME = "newInstance";
    private static HashMap<AssemblePush, ManageClassInfo> mHashMaps = new HashMap<>();

    /* JADX INFO: renamed from: com.xiaomi.mipush.sdk.AssemblePushInfoHelper$1, reason: invalid class name */
    /* JADX INFO: loaded from: miuipushsdkshared_3_7_9.jar:com/xiaomi/mipush/sdk/AssemblePushInfoHelper$1.class */
    static /* synthetic */ class AnonymousClass1 {
        static final /* synthetic */ int[] $SwitchMap$com$xiaomi$mipush$sdk$AssemblePush;

        static {
            int[] iArr = new int[AssemblePush.values().length];
            $SwitchMap$com$xiaomi$mipush$sdk$AssemblePush = iArr;
            try {
                iArr[AssemblePush.ASSEMBLE_PUSH_HUAWEI.ordinal()] = 1;
            } catch (NoSuchFieldError e) {
            }
            try {
                $SwitchMap$com$xiaomi$mipush$sdk$AssemblePush[AssemblePush.ASSEMBLE_PUSH_FCM.ordinal()] = 2;
            } catch (NoSuchFieldError e2) {
            }
            try {
                $SwitchMap$com$xiaomi$mipush$sdk$AssemblePush[AssemblePush.ASSEMBLE_PUSH_COS.ordinal()] = 3;
            } catch (NoSuchFieldError e3) {
            }
            try {
                $SwitchMap$com$xiaomi$mipush$sdk$AssemblePush[AssemblePush.ASSEMBLE_PUSH_FTOS.ordinal()] = 4;
            } catch (NoSuchFieldError e4) {
            }
        }
    }

    /* JADX INFO: loaded from: miuipushsdkshared_3_7_9.jar:com/xiaomi/mipush/sdk/AssemblePushInfoHelper$ManageClassInfo.class */
    static class ManageClassInfo {
        public String className;
        public String methodName;

        public ManageClassInfo(String str, String str2) {
            this.className = str;
            this.methodName = str2;
        }
    }

    static {
        add(AssemblePush.ASSEMBLE_PUSH_HUAWEI, new ManageClassInfo(HMS_PUSH_MANAGER_CLASS_NAME, "newInstance"));
        add(AssemblePush.ASSEMBLE_PUSH_FCM, new ManageClassInfo(FCM_PUSH_MANAGER_CLASS_NAME, "newInstance"));
        add(AssemblePush.ASSEMBLE_PUSH_COS, new ManageClassInfo(COS_PUSH_MANAGER_CLASS_NAME, "newInstance"));
        add(AssemblePush.ASSEMBLE_PUSH_FTOS, new ManageClassInfo(FTOS_PUSH_MANAGER_CLASS_NAME, "newInstance"));
    }

    private static void add(AssemblePush assemblePush, ManageClassInfo manageClassInfo) {
        if (manageClassInfo != null) {
            mHashMaps.put(assemblePush, manageClassInfo);
        }
    }

    public static ConfigKey getConfigKeyByType(AssemblePush assemblePush) {
        return ConfigKey.AggregatePushSwitch;
    }

    public static ManageClassInfo getManageClassInfoByType(AssemblePush assemblePush) {
        return mHashMaps.get(assemblePush);
    }

    public static RetryType getRetryType(AssemblePush assemblePush) {
        RetryType retryType;
        switch (AnonymousClass1.$SwitchMap$com$xiaomi$mipush$sdk$AssemblePush[assemblePush.ordinal()]) {
            case 1:
                retryType = RetryType.UPLOAD_HUAWEI_TOKEN;
                break;
            case 2:
                retryType = RetryType.UPLOAD_FCM_TOKEN;
                break;
            case 3:
                retryType = RetryType.UPLOAD_COS_TOKEN;
                break;
            case 4:
                retryType = RetryType.UPLOAD_FTOS_TOKEN;
                break;
            default:
                retryType = null;
                break;
        }
        return retryType;
    }
}

File: legacy-runtime/src/main/java/com/xiaomi/mipush/sdk/AssemblePushInfoHelper.kt
package com.xiaomi.mipush.sdk

import com.xiaomi.xmpush.thrift.ConfigKey
import java.util.HashMap

/*
 * Current override reference: com.xiaomi.xmsf 0.3.17-20260410000745 (versionCode 1003003000),
 * base.apk sha256 f3d72b6f5e1427ceecd3147a051d58e4dc95bb528397d486658e01cad9f7e590,
 * JADX path: com.xiaomi.xmsf/current/base/sources/com/xiaomi/mipush/sdk/AssemblePushInfoHelper.java
 * No stock 7.4.67-C same-path source was found in the split source tree.
 */
abstract class AssemblePushInfoHelper {
    class ManageClassInfo(
        @JvmField var className: String?,
        @JvmField var methodName: String?
    )

    companion object {
        protected const val COS_PUSH_MANAGER_CLASS_NAME = "com.xiaomi.assemble.control.COSPushManager"
        protected const val COS_PUSH_SINGLE_METHOD_NAME = "newInstance"
        protected const val FCM_PUSH_MANAGER_CLASS_NAME = "com.xiaomi.assemble.control.FCMPushManager"
        protected const val FCM_PUSH_SINGLE_METHOD_NAME = "newInstance"
        protected const val FTOS_PUSH_MANAGER_CLASS_NAME = "com.xiaomi.assemble.control.FTOSPushManager"
        protected const val FTOS_PUSH_SINGLE_METHOD_NAME = "newInstance"
        protected const val HMS_PUSH_MANAGER_CLASS_NAME = "com.xiaomi.assemble.control.HmsPushManager"
        protected const val HMS_PUSH_SINGLE_METHOD_NAME = "newInstance"

        private val mHashMaps = HashMap<AssemblePush, ManageClassInfo>()

        init {
            add(AssemblePush.ASSEMBLE_PUSH_HUAWEI, ManageClassInfo(HMS_PUSH_MANAGER_CLASS_NAME, HMS_PUSH_SINGLE_METHOD_NAME))
            add(AssemblePush.ASSEMBLE_PUSH_FCM, ManageClassInfo(FCM_PUSH_MANAGER_CLASS_NAME, FCM_PUSH_SINGLE_METHOD_NAME))
            add(AssemblePush.ASSEMBLE_PUSH_COS, ManageClassInfo(COS_PUSH_MANAGER_CLASS_NAME, COS_PUSH_SINGLE_METHOD_NAME))
            add(AssemblePush.ASSEMBLE_PUSH_FTOS, ManageClassInfo(FTOS_PUSH_MANAGER_CLASS_NAME, FTOS_PUSH_SINGLE_METHOD_NAME))
        }

        private fun add(assemblePush: AssemblePush, manageClassInfo: ManageClassInfo?) {
            if (manageClassInfo != null) {
                mHashMaps[assemblePush] = manageClassInfo
            }
        }

        @JvmStatic
        fun getConfigKeyByType(assemblePush: AssemblePush?): ConfigKey {
            return ConfigKey.AggregatePushSwitch
        }

        @JvmStatic
        fun getManageClassInfoByType(assemblePush: AssemblePush?): ManageClassInfo? {
            return mHashMaps[assemblePush]
        }

        @JvmStatic
        fun getRetryType(assemblePush: AssemblePush?): RetryType? {
            return when (assemblePush) {
                AssemblePush.ASSEMBLE_PUSH_HUAWEI -> RetryType.UPLOAD_HUAWEI_TOKEN
                AssemblePush.ASSEMBLE_PUSH_FCM -> RetryType.UPLOAD_FCM_TOKEN
                AssemblePush.ASSEMBLE_PUSH_COS -> RetryType.UPLOAD_COS_TOKEN
                AssemblePush.ASSEMBLE_PUSH_FTOS -> RetryType.UPLOAD_FTOS_TOKEN
                else -> null
            }
        }
    }
}
